### PR TITLE
Mapped out Enchantment and related classes

### PR DESCRIPTION
**Pull request by Darkhax**
This PR maps the Enchantment class further, and also maps related classes. Things like implementations, or classes that are used/referenced by the Enchantment class.

### CODE REVIEW

File: mappings/net/minecraft/enchantment/Enchantment.mapping [hunk 1]
```diff
@@ -1,17 +1,53 @@
 CLASS none/ahm net/minecraft/enchantment/Enchantment
+	CLASS none/ahm$a Weight
+		FIELD a COMMON Lnone/ahm$a;
+		FIELD b UNCOMMON Lnone/ahm$a;
+		FIELD c RARE Lnone/ahm$a;
+		FIELD d LEGENDARY Lnone/ahm$a;
```
Comment (darkevilmac) [inline mappings/net/minecraft/enchantment/Enchantment.mapping:6]: Maybe the weights should be named in a more standard way like: EXTREME, HIGH, MEDIUM, LOW.

Comment (Darkhax) [inline mappings/net/minecraft/enchantment/Enchantment.mapping:6]: In this case the weighting is used to describe the rarity of the enchantment.

Comment (asiekierka) [inline mappings/net/minecraft/enchantment/Enchantment.mapping:6]: I can't believe you used LEGENDARY. Pfft. It's fine, though.

File: mappings/net/minecraft/enchantment/Enchantment.mapping [hunk 2]
```diff
@@ -1,17 +1,53 @@
...
+	METHOD a onTargetDamaged (Lnone/sn;Lnone/se;I)V
+		ARG 0 user
+		ARG 1 target
+		ARG 2 level
+	METHOD b getHighestLevel ()I
 	METHOD b getEnchantmentById (Ljava/lang/String;)Lnone/ahm;
 		ARG 0 id
 	METHOD b getRawIdFromEnchantment (Lnone/ahm;)I
 		ARG 0 enchantment
+	METHOD b onUserDamaged (Lnone/sn;Lnone/se;I)V
```
Comment (asiekierka) [inline mappings/net/minecraft/enchantment/Enchantment.mapping:40]: I assume those are reactive and not active? (That is, they don't actually damage things, just react to a damage happening)

**Comment by asiekierka** [comment]
Other than that question, LGTM

File: mappings/net/minecraft/enchantment/Enchantment.mapping (hunk 2 above)
Comment (copygirl) [inline mappings/net/minecraft/enchantment/Enchantment.mapping:40]: `User` is a bit generic. Is this just for armor? I'd suggest `Wearer`, even if it sounds a bit odd.

Comment (shadowfacts) [inline mappings/net/minecraft/enchantment/Enchantment.mapping:40]: This method is for armor only right? If so, I concur with @copygirl, Wearer would make more sense.

**Comment by shadowfacts** [comment]
Other than my one comment, this looks good to me.

File: mappings/net/minecraft/enchantment/Enchantment.mapping (hunk 2 above)
Comment (Darkhax) [inline mappings/net/minecraft/enchantment/Enchantment.mapping:40]: This is actually not limited to armor. Enchantments like Thorns use this method for their effect. While thorns is an armor enchantment, the effect also works if the enchantment is on the held item, or the offhand item.

**Comment by asiekierka** [comment]
LGTM

File: mappings/net/minecraft/enchantment/Enchantment.mapping (hunk 2 above)
Comment (copygirl) [inline mappings/net/minecraft/enchantment/Enchantment.mapping:40]: How about `Owner`?

**Pull request merged**